packages/test-runner/src/fixtures.ts

```diff
@@ -116,30 +121,52 @@ class Fixture {
   }
 }
 
+class IncrementableArray<T> {
```

Comment:
It looks like we are trying to address unusual design with an unusual data structure. Let's rather bring it back into a shape where it is easy to understand and maintain!

Let's build a tree of registrations and resolve / instantiate fixtures based on this tree. We should end up in a state where there can be two fixtures named 'A' and one depends on another (overrides it). And both have fixture instances, cached values, following the registration tree.

We will then rerun the registrations for each new test to re-initialize it every time we run a new test.